lib/apr/events.ex

```diff
@@ -147,8 +151,8 @@ defmodule Apr.Events do
     case fetch_response do
       {:ok, response} ->
         artworks_map =
-          response.body["data"]["artworks"]
-          |> Enum.reduce(%{}, fn a, acc -> Map.merge(acc, %{a["_id"] => a}) end)
+          response.body["data"]["artworks"]["edges"]
+          |> Enum.reduce(%{}, fn a, acc -> Map.merge(acc, %{a["node"]["internalID"] => a}) end)
```

Comment:
Okay I have to do some Elixir refreshing in order to understand what is happening here :) So `Enum.reduce/3` is taking a Enum, a function and an accumulator
```elixir
Enum.reduce([1, 1, 2, 2], fn x, acc -> x + acc end) => 6
```
And `Map.merge/2` is merging 2 maps
```elixir
Map.merge(%{kaja: 21, jackie: 13}, %{joey: 42})
=> %{jackie: 13, joey: 42, kaja: 21}
```
So here we are stuffing the return value of the response body (a map) into the `Enum.reduce` that merges another map with the accumulator and assigns is to the original map?

Reply:
I would say in order to write a test for the changes, I would refactor the `fetch_artworks(order_events)` function (cause it is very long) and extract parts into smaller functions that are partially testable. For example `fetch_response` could be extracted into its own function and also `artworks_map` could be extracted into its own function and like that they would both be easier to test with some test value.

Reply:
Maybe a test for this could be in events_test.exs and look something like
```elixir
test "fetch_artworks/1 returns artworks" do
  order_events = [Fixtures.create(:event, @valid_attrs)]
  assert Events.fetch_artworks(order_events) == [order_events]
end
```

Reply:
No I don't but I would be happy to find out together in a session! I guess this would mean running a graphQL query locally and getting a hash with these properties back to see what APRD does with them. I was locally trying out some refactoring changes and running tests with some `|> IO.inspect()` in the code to see what the return values of some of this is. If we have a fixture in the tests that looks like an expected graphQL response, we could at least see in the test environment that the code itself does not do anything unexpected. (My insecurity here comes from my own lack of knowledge of Elixir, not from the lack of trust in your implementation)

Reply:
As a sort of MVP, this is just updating the field names from MPv1 to MPv2, which looks good from over here